media-tv/kodi/kodi-20.9999.ebuild

```diff
@@ -0,0 +1,462 @@
+	')
+	airplay? (
+		>=app-pda/libplist-2.0.0:=
+		net-libs/shairplay
+	)
+	alsa? (
+		>=media-libs/alsa-lib-1.1.4.1
+	)
+	bluetooth? (
+		net-wireless/bluez:=
+	)
+	bluray? (
+		>=media-libs/libbluray-1.1.2:=
+	)
+	caps? (
+		sys-libs/libcap
+	)
+	cec? (
+		>=dev-libs/libcec-4.0[-cubox,raspberry-pi?]
+	)
+	dbus? (
+		sys-apps/dbus
+	)
+	gbm? (
+		>=dev-libs/libinput-1.10.5:=
+		x11-libs/libxkbcommon
+	)
+	gles? (
+		!raspberry-pi? (
+			media-libs/mesa[gles2]
+		)
+	)
+	!gles? (
+		media-libs/glu
+	)
+	lcms? (
+		>=media-libs/lcms-2.10:2
+	)
+	libusb? (
+		virtual/libusb:1
+	)
+	lirc? (
+		app-misc/lirc
+	)
+	mariadb? (
+		dev-db/mariadb-connector-c:=
+	)
+	mysql? (
+		dev-db/mysql-connector-c:=
+	)
+	nfs? (
+		>=net-fs/libnfs-2.0.0:=
+	)
+	pipewire? (
+		>=media-video/pipewire-0.3.24:=
+	)
+	pulseaudio? (
+		media-libs/libpulse
+	)
+	raspberry-pi? (
+		|| (
+			media-libs/raspberrypi-userland
+			media-libs/raspberrypi-userland-bin
+			media-libs/mesa[gles2,video_cards_vc4]
+		)
+	)
+	samba? (
+		>=net-fs/samba-3.4.6[smbclient(+)]
+	)
+	system-ffmpeg? (
+		>=media-video/ffmpeg-${FFMPEG_VERSION}:=[encode,postproc,vaapi?,vdpau?,X?]
+		=media-video/ffmpeg-4*[openssl]
+	)
+	!system-ffmpeg? (
+		app-arch/bzip2
+		media-libs/dav1d:=
+	)
+	udf? (
+		>=dev-libs/libudfread-1.0.0
+	)
+	udev? (
+		virtual/libudev:=
+	)
+	vaapi? (
+		media-libs/libva:=[wayland?,X?]
+	)
+	vdpau? (
+		|| (
+			>=x11-libs/libvdpau-1.1
+			>=x11-drivers/nvidia-drivers-180.51
+		)
+	)
+	wayland? (
+		>=x11-libs/libxkbcommon-0.4.1[wayland]
+	)
+	webserver? (
+		>=net-libs/libmicrohttpd-0.9.55:=[messages(+)]
+	)
+	X? (
+		x11-libs/libX11
+		x11-libs/libXext
+		x11-libs/libXrandr
+	)
+	xslt? (
+		dev-libs/libxslt
+		>=dev-libs/libxml2-2.9.4
+	)
+	zeroconf? (
+		net-dns/avahi[dbus]
+	)
+"
+RDEPEND="
+	${COMMON_DEPEND}
+	${COMMON_TARGET_DEPEND}
+"
+DEPEND="
+	${COMMON_DEPEND}
+	${COMMON_TARGET_DEPEND}
+	dev-libs/rapidjson
+	test? (
+		>=dev-cpp/gtest-1.10.0
+	)
+	wayland? (
+		>=dev-libs/wayland-protocols-1.7
+	)
+	X? (
+		x11-base/xorg-proto
+		x11-libs/libXrender
+	)
+"
+BDEPEND="
+	${COMMON_DEPEND}
+	dev-lang/swig
+	dev-util/cmake
+	virtual/pkgconfig
+	virtual/jre
+	doc? (
+		app-doc/doxygen
+	)
+"
+
+# bug #544020
+CONFIG_CHECK="~IP_MULTICAST"
+ERROR_IP_MULTICAST="
+In some cases Kodi needs to access multicast addresses.
+Please consider enabling IP_MULTICAST under Networking options.
+"
+
+PATCHES=(
+	"${FILESDIR}"/kodi-20.2-binutils-2.41.patch
+)
+
+pkg_setup() {
+	check_extra_config
+	python-single-r1_pkg_setup
+}
+
+src_unpack() {
+	if [[ ${PV} == *9999 ]] ; then
+		git-r3_src_unpack
+	else
+		unpack ${MY_P}.tar.gz
+	fi
+}
+
+src_prepare() {
+	cmake_src_prepare
+
+	# avoid long delays when powerkit isn't running #348580
+	sed -i \
+		-e '/dbus_connection_send_with_reply_and_block/s:-1:3000:' \
+		xbmc/platform/linux/*.cpp || die
+
+	# fix finding kissfft[openmp]
+	sed -i \
+		-e 's/\(find_library(KISSFFT_LIBRARY NAMES .*\)/\1 kissfft-simd-openmp/' \
```

Comment:
This doesn't work on arm or arm64. It's `kissfft-float-openmp` on those. Upstream should probably be told to add openmp here.

Reply:
I wonder if I could force it to use the cmake config for finding it.